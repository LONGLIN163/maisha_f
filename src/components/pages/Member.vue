<template>
    <div>
        <div>
            <van-nav-bar  title="Member" />
        </div>
        <div class="top">
           <img src="https://i.pinimg.com/originals/0c/3b/3a/0c3b3adb1a7530892e55ef36d3be6cb8.png" class="top-img" />
        </div>
        <div class="login">
            <div><van-button data-test="loginbtn" type="warning" @click="goToLogin">Sign in</van-button></div>
            <div><van-button  data-test="signupbtn" type="primary" @click="goToRegister">Singn up</van-button></div>
        </div>
        <div>
            <van-cell-group>
                <van-cell title="Card" is-link />
                <van-cell title="Address" is-link  />
                <van-cell title="Orders" is-link  />
                <van-cell title="Member rights" is-link />
                <van-cell title="Contact" is-link  />
            </van-cell-group>

        </div>

    </div>
</template>

<script>
    export default {
        data() {
            return {
                
            }
        },
        methods:{
            goToRegister(){
                this.$router.push({name:'Register'})
            },
            goToLogin(){
                this.$router.push({name:'Login'})
            },
        }
    }
</script>

<style lang="scss" scoped>
    .top{
        height:5rem;
        text-align: center;
        padding-top:2rem;
        background-color: $color-primary;
        .top-img{
            width:70px;
            height: 70px;
            border-radius: 50px;
        }
    }
    .login{
        display: flex;
        flex-direction: row;
        background-color: $color-background-tape;
        padding:10px;
        div{
            flex:1;
            text-align: center;
        }
    }
</style>