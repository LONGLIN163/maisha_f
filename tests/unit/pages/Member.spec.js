import {mount, createLocalVue} from '@vue/test-utils'
import Member from '@/components/pages/Member.vue'

import Vue from 'vue'

const Vant=require("vant")

const {Field}=Vant

Vue.use(Field)

const localVue = createLocalVue();
localVue.use(Vant); 



describe('Member', () => {
  const wrapper = mount(Member,{
    localVue,
  })

  it('renders a div', () => {
    expect(wrapper).toMatchSnapshot()
  })
})