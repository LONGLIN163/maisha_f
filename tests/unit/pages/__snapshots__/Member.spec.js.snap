// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member renders a div 1`] = `
<div>
  <div>
    <div class="van-nav-bar van-hairline--bottom">
      <div class="van-nav-bar__content">
        <div class="van-nav-bar__title van-ellipsis">Member</div>
      </div>
    </div>
  </div>
  <div class="top"><img src="https://i.pinimg.com/originals/0c/3b/3a/0c3b3adb1a7530892e55ef36d3be6cb8.png" class="top-img"></div>
  <div class="login">
    <div><button class="van-button van-button--warning van-button--normal">
        <div class="van-button__content"><span class="van-button__text">Sign in</span></div>
      </button></div>
    <div><button class="van-button van-button--primary van-button--normal">
        <div class="van-button__content"><span class="van-button__text">Singn up</span></div>
      </button></div>
  </div>
  <div>
    <div class="van-cell-group van-hairline--top-bottom">
      <div role="button" tabindex="0" class="van-cell van-cell--clickable">
        <div class="van-cell__title"><span>Card</span></div><i class="van-icon van-icon-arrow van-cell__right-icon">
          <!---->
        </i>
      </div>
      <div role="button" tabindex="0" class="van-cell van-cell--clickable">
        <div class="van-cell__title"><span>Address</span></div><i class="van-icon van-icon-arrow van-cell__right-icon">
          <!---->
        </i>
      </div>
      <div role="button" tabindex="0" class="van-cell van-cell--clickable">
        <div class="van-cell__title"><span>Orders</span></div><i class="van-icon van-icon-arrow van-cell__right-icon">
          <!---->
        </i>
      </div>
      <div role="button" tabindex="0" class="van-cell van-cell--clickable">
        <div class="van-cell__title"><span>Member rights</span></div><i class="van-icon van-icon-arrow van-cell__right-icon">
          <!---->
        </i>
      </div>
      <div role="button" tabindex="0" class="van-cell van-cell--clickable">
        <div class="van-cell__title"><span>Contact</span></div><i class="van-icon van-icon-arrow van-cell__right-icon">
          <!---->
        </i>
      </div>
    </div>
  </div>
</div>
`;
